### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**
[DEMO LINK](https://ihorandrianov.github.io/react_movies-list-add-form/)

### CODE REVIEW

**Review by naziamoff — CHANGES_REQUESTED** [review]
- You should clear the form fields after the submit
- I can add one movie a lot of times, you should validate it
- I can put two spaces as a movie name. You should handle this too

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
+
+  const handleAdd = (movie: Movie) => {
+    setMovies((prevState) => {
+      const newState = [...prevState];
+
+      newState.push(movie);
```
Comment (Denchepornyuk) [inline src/App.tsx:15]: don't use push, add movie with array copy

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { FC } from 'react';
+import { useFormikContext } from 'formik';
+import { Movie } from '../types/Movie';
+
+export const Button: FC = () => {
+  const ctx = useFormikContext<Movie>();
```
Comment (Denchepornyuk) [inline src/components/Button.tsx:6]: 🔥

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,58 +1,92 @@
...
+          .required(),
+        imdbId: Yup.string()
+          .required()
+          .whitespace('Write some id here!'),
+      })}
+      onSubmit={(values, { resetForm }) => {
+        onAdd(values);
+        resetForm();
+      }}
+    >
```
Comment (Denchepornyuk) [inline src/components/NewMovie/NewMovie.tsx:59]: I didn't work with formik, we use React Final Form in Mate but it looks great 🚀

**Review by Denchepornyuk — APPROVED** [review]

**Review by naziamoff — APPROVED** [review]

**Pull request closed without merging**